src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -209,6 +193,22 @@ private boolean acceptOrRejectTask(Task task, String walletAddress) {
         }
 
         try {
+            final Optional<ReplicatesList> oReplicatesList = replicatesService.getReplicatesList(chainTaskId);
+            // Check is only here to prevent
+            // "`Optional.get()` without `isPresent()` warning".
+            // This case should not happen.
+            if (oReplicatesList.isEmpty()) {
+                return false;
+            }
+
+            final ReplicatesList replicatesList = oReplicatesList.get();
```

Comment:
Maybe this more condensed form to avoid unnecessary `final Optional` ?
```suggestion
            final ReplicatesList replicatesList = replicatesService.getReplicatesList(chainTaskId).orElse(null);

            if (replicatesList == null) {
                return false;
            }
```

Reply:
This can totally be done at a later time or not at all

Reply:
You're right we can do something better. I've tried the functional way. Let me know if it suits you :)
https://github.com/iExecBlockchainComputing/iexec-core/pull/644/commits/5ddcf26f978f3a59c78fdc11f939c263c06ce419